Userland/Applications/PixelPaint/ImageEditor.cpp

```diff
@@ -203,6 +275,11 @@ void ImageEditor::mousedown_event(GUI::MouseEvent& event)
 
 void ImageEditor::mousemove_event(GUI::MouseEvent& event)
 {
+    auto old_mouse_position = m_mouse_position;
+    m_mouse_position = event.position();
+    if (m_mouse_position != old_mouse_position)
+        update(); // TODO: Limit to indicator rects
```

Comment:
Yeah I don't think it makes sense to be updating the entire image editor on every mouse move event, especially since andress has been careful to not redraw more than necessary. Perhaps making these a separate widget would help with this.

Reply:
I feel it's in the middle ground between it belonging to the `ImageEditor` class and being its own class. I think I saw an overload for `update` that took a rect that limited the update, so that was my initial plan, but I'll see if ki can refactor it to it's own class.

Reply:
Yeah I agree, it's not exactly clear where it should reside. I was just suggesting another pro of doing it as a separate widget. Maybe someone else can chip in with a more useful opinion on that :^)

Reply:
@mustafaquraish Check out the spacing between the image and border in the PR description. To get it to center correctly we need to hack `ImageEditor::reset_scale_and_position` and potentially other functions. I guess it would be most correct to refactor it out to its own widget.

@awesomekling - What's your take on keeping the rulers as a part of `ImageEditor` (by just adding draw calls in `paint_event` as it currently is) or making them as separate widgets?